### PR TITLE
core: sync BaseModel and FastCamelContext with Camel main

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@ppalaga @jamesnetherton more tests may be required, this is the bare minimum I got with the time I had today.
@davsclaus mind checking if I missed something ?

**Review by nicolaferraro — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

File: integration-tests/kamelet/src/test/java/org/apache/camel/quarkus/component/kamelet/it/KameletTest.java [hunk 1]
```diff
@@ -65,4 +65,23 @@ public void testKameletChain() {
...
+                .contentType(ContentType.TEXT)
+                .body("Kamelet")
+                .post("/kamelet/invoke/toUpperWithBean")
+                .then()
+                .statusCode(200)
+                .body(is("KAMELET"));
+
+        RestAssured.given()
+                .contentType(ContentType.TEXT)
+                .body("Kamelet")
```
Comment (davsclaus) [inline integration-tests/kamelet/src/test/java/org/apache/camel/quarkus/component/kamelet/it/KameletTest.java:81]: I would send another input body so the result is different from the previous test, eg such as Kamelet2

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
Nice to see all this non functional code ;)

File: integration-tests/kamelet/src/test/java/org/apache/camel/quarkus/component/kamelet/it/KameletTest.java (hunk 1 above)
Comment (lburgazzoli) [reply integration-tests/kamelet/src/test/java/org/apache/camel/quarkus/component/kamelet/it/KameletTest.java]: done

**Review by jamesnetherton — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="UTF-8"?>
+﻿<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ppalaga) [inline pom.xml:1]: Looks like you have added a byte order mark. Any reason for that?

```
$ hexdump -n 3 -C pom.xml 
00000000  ef bb bf                                          |...|
```

before the change:

```
$ hexdump -n 3 -C pom.xml 
00000000  3c 3f 78                                          |<?x|
```

**Comment by lburgazzoli** [comment]
no reason

**Comment by ppalaga** [comment]
Thanks, let me revert the change
